messaging.rst

```diff
@@ -9,125 +9,6 @@ This is the specification document for the messages used in the Raiden protocol.
-+------------------------+------------+--------------------------------------------------------------------------------+
-
-.. _cooperative-settle-proof-message:
-
-Cooperative Settle Proof
-------------------------
-
-Data required by the smart contracts to allow the two channel participants to close and settle the channel instantly, in one transaction.
-Signature must be valid and is defined as:
-
-::
-
-    ecdsa_recoverable(privkey, sha3_keccak(participant1_address || participant1_balance || participant2_address || participant2_balance || channel_identifier || token_network_address || chain_id)
-
-Fields
-^^^^^^
-
-+------------------------+------------+--------------------------------------------------------------------------------+
-| Field Name             | Field Type |  Description                                                                   |
-+========================+============+================================================================================+
-|  participant1_address  | address    | One of the channel participants                                                |
-+------------------------+------------+--------------------------------------------------------------------------------+
-|  participant1_balance  | uint256    | Amount of tokens that participant1_address will receive after settling         |
-+------------------------+------------+--------------------------------------------------------------------------------+
-|  participant2_address  | address    | The other channel participant                                                  |
-+------------------------+------------+--------------------------------------------------------------------------------+
-|  participant2_balance  | uint256    | Amount of tokens that participant2_address will receive after settling         |
-+------------------------+------------+--------------------------------------------------------------------------------+
-|  channel_identifier    | uint256    | Channel identifier inside the TokenNetwork contract                            |
-+------------------------+------------+--------------------------------------------------------------------------------+
-| token_network_address  | address    | Address of the TokenNetwork contract                                           |
-+------------------------+------------+--------------------------------------------------------------------------------+
-| chain_id               | uint256    | Chain identifier as defined in EIP155                                          |
-+------------------------+------------+--------------------------------------------------------------------------------+
-|  signature             | bytes      | Elliptic Curve 256k1 signature on the above data                               |
-+------------------------+------------+--------------------------------------------------------------------------------+
-
```

Comment:
In the `Data Structures` section I would leave the following data structures:
- Balance Proof - defined as https://github.com/raiden-network/raiden/blob/master/raiden/transfer/state.py#L459, in a similar format as defined initially (see 
 also https://github.com/raiden-network/spec/blob/7134b3fce660c85b132f2e6e3f182715cd45a32e/messaging.rst#balance-proof), but with up-to-date fields.

Then, reference it the `Messages` as needed, without repeating the fields.
Balance proof exists as a concept both in the smart contracts and the Raiden client.

Reply:
To move the balance proof definition to the smart contracts session was an explicit todo in issue #85.

Reply:
Yes, the issue description was not clear enough.
The problem was that the balance proof definition from `messages.py` is not correct from the Raiden client perspective, but it is correct from the smart contracts perspective -> should be moved to the smart contracts.

As mentioned in the issue:
> add the actual data structure used by the messages

For `messages.py` we still need to define this Balance Proof data structure, but with the correct components from the Raiden client perspective - e.g. have `transferred`, `locked_amount`, `locksroot` instead of the contract's `balance_hash`.
So, I failed to add a reference to what needs to be used in the spec here, which is: https://github.com/raiden-network/raiden/blob/1ca9e251fe33564a430c06d9a9b49614daa7b916/raiden/transfer/state.py#L466-L482

Now, this was my take on it. Maybe someone else has another opinion on the specs.

Reply:
Ok, I'll change the `Messages` section accordingly. I'll also revert the two single-line changes you commented below; if it is not about individual data structures for the different messages the distinction between balance proof and data to update the balance proof is not so important.